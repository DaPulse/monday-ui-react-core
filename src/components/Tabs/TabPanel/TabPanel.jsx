import cx from "classnames";
import React, { forwardRef, useRef } from "react";
import PropTypes from "prop-types";
import useMergeRefs from "../../../hooks/useMergeRefs";
import styles from "./TabPanel.module.scss";
import { ELEMENT_TYPES, getTestId } from "../../../utils/test-utils";

const TabPanel = forwardRef(({ className, id, children, index, dataTestId }, ref) => {
  const componentRef = useRef(null);
  const mergedRef = useMergeRefs({ refs: [ref, componentRef] });

  return (
    <div
      key={`${id}_${index}`}
      ref={mergedRef}
      className={cx(styles.tabPanelWrapper, "tab-panel--wrapper", className)}
      id={id}
      role="tabpanel"
      data-testid={dataTestId || getTestId(ELEMENT_TYPES.TAB_PANEL)}
    >
      {children}
    </div>
  );
});

TabPanel.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string
};
TabPanel.defaultProps = {
  className: "",
  id: ""
};

export default TabPanel;
