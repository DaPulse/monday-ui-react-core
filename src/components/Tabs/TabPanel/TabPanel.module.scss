@import "../../../styles/themes";
@import "../../../styles/typography";

.tabPanelWrapper {
  color: var(--primary-text-color);
}
