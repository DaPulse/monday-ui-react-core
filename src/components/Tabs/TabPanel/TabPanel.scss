@import "../../../styles/themes";
@import "../../../styles/typography";

.tab-panel--wrapper {

}
