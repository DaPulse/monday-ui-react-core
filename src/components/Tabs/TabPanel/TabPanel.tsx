import cx from "classnames";
import React, { FC, forwardRef, useRef } from "react";
import useMergeRefs from "../../../hooks/useMergeRefs";
import VibeComponentProps from "../../../types/VibeComponentProps";
import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import styles from "./TabPanel.module.scss";

export interface TabPanelProps extends VibeComponentProps {
  children?: string;
  index?: number;
}

const TabPanel: FC<TabPanelProps> = forwardRef(({ className, id, children, index, "data-testid": dataTestId }, ref) => {
  const componentRef = useRef(null);
  const mergedRef = useMergeRefs({ refs: [ref, componentRef] });

  return (
    <div
      key={`${id}_${index}`}
      ref={mergedRef}
      className={cx(styles.tabPanelWrapper, "tab-panel--wrapper", className)}
      id={id}
      data-testid={dataTestId || getTestId(ComponentDefaultTestId.TAB_PANEL, id)}
      role="tabpanel"
    >
      {children}
    </div>
  );
});

export default TabPanel;
