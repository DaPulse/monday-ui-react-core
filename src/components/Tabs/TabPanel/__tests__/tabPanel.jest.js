import React from "react";
import renderer from "react-test-renderer";
import TabPanel from "../TabPanel";

it("renders correctly with empty props", () => {
  const tree = renderer.create(<TabPanel />).toJSON();
  expect(tree).toMatchSnapshot();
});
