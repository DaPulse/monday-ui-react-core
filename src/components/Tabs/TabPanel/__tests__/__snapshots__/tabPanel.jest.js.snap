// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with empty props 1`] = `
<div
  className="tab-panel--wrapper"
  id=""
  role="tabpanel"
/>
`;
